### ISSUE TITLE
Add hasName to speed limits quest?

### ISSUE THREAD

**Issue by goldfndr** [westnordost/StreetComplete#975]
I was taking StreetComplete 4.1 out for a time, and this was the first time I used it for speed limits (keeping mindful of #813 in USA) and road surfaces. I was a bit surprised that, while the road name was mentioned in the road surface quest, it wasn't in the speed limit quest.

Would a solution be as simple as adding a hasName test and another string? (I'd try it myself, but I'm a bit discouraged by the 1.5 hour build time.)

**Comment by matkoniecz** [comment]
> 1.5 hour build time

Is it really, really, really old computer or broken system (for example old unsupported Windows with miner malware?)?

Because it sounds like something going wrong - I have very old laptop (Lenovo G550) and SC builds within minutes.

**Comment by rugk** [comment]
> old unsupported Windows

Yes, see  [previous discussion](https://github.com/westnordost/StreetComplete/issues/950#issuecomment-370915138).

**Comment by rugk** [comment]
BTW upgrading to Android Studio 3.0 won't hurt. 😄

**Comment by goldfndr** [comment]
Android Studio 2.3.3 is the last version compatible with Windows XP, upgrading to Android Studio 3.* can't be performed.

**Comment by rugk** [comment]
Generally, BTW, you may have a look at switching the OS. You would not want to use WinXP nowadays, alone the missing security updates. Many Linux distro's or so are likely less resource-heavy and you can install Android Studio 3.0, BTW.

(But it's getting of-topic here…)

**Comment by westnordost** [comment]
> (keeping mindful of #813 in USA) 

You built a custom version just to re-enable the maxspeed quest in your area? Phew. You know, I am interested to enable this quest as well in the US, but I do not feel entitled to define the necessary tagging scheme for that. Isn't there some active (US) community members who would be willing to discuss and come up with somethin in the forum/mailing list/other?

Regarding the feature proposal: Would probably make sense. The reason why it was implemented for the surface quest is to better refer to it in the undo-dialog. I guess it is the same with this quest. Though, a better UI for undoing would also solve this (i.e. showing the affected geometry next to a list of undoable quests).

**Comment by goldfndr** [comment]
I didn't build a custom version for maxspeed in US. I simply didn't invalidate the cache of downloaded quests in the released version. I think I did try copying the databases (using a root explorer) to the debug version I built to test the library bug and it crashed at startup (no GUI) so I ran the debug clean. With Android Studio 2.3.3, the debug vs release toggle isn't displaying.

Here in Washington state, there are at least 13 cities in King County, each with their own unsigned speed limits that can differ from the county's unsigned speed limits. The US has something like 19500 cities and 3000 counties, although I suppose some percentage of cities aren't differing from the county's values (and thus wouldn't need their own polygons). Roads aren't split at city boundaries, in my experience.

**Comment by goldfndr** [comment]
Well, maybe there isn't as much of a collection of fiefdoms as I thought, maybe the cities are posting citywide speed limits that happen to match the state limits in an effort to remind, not to override. I'll try to do some more spot checking.

**Comment by westnordost** [comment]
The app does not need to know the unsigned speed limits to tag correctly but only which factors contribute to which implicit speed limits apply (i.e. built-up area vs rural area)

On 15 March 2018 01:12:22 CET, Richard Finegold <notifications@github.com> wrote:
>I didn't build a custom version for maxspeed in US. I simply didn't
>invalidate the cache of downloaded quests in the released version. I
>think I did try copying the databases (using a root explorer) to the
>debug version I built to test the library bug and it crashed at startup
>(no GUI) so I ran the debug clean. With Android Studio 2.3.3, the debug
>vs release toggle isn't displaying.
>
>Here in Washington state, there are at least 13 cities in King County,
>each with their own unsigned speed limits that can differ from the
>county's unsigned speed limits. The US has something like 19500 cities
>and 3000 counties, although I suppose some percentage of cities aren't
>differing from the county's values (and thus wouldn't need their own
>polygons). Roads aren't split at city boundaries, in my experience.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/975#issuecomment-373217043

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by gappleto97** [comment]
In regards to custom versions, I'm confused. I have the F-Droid version and have invalidated my quest cache multiple times since the new release, but I still have a lot of speed limit quests. I like that I still have them, but it sounds like this was not the intent. Can you confirm?

**Comment by ENT8R** [comment]
Invalidating the quest cache does **not** mean that the quests get deleted... The area of the quests is just marked as not downloaded after you invalidated your quest cache. So you have to redownload all quests again in that area...

**Comment by gappleto97** [comment]
It also asked these questions when I went to areas that I had never used this app in before. For instance, it asked them when I was in Las Vegas, but I had never opened the app in that area before. Same thing for Minneapolis.

**Comment by ENT8R** [comment]
Well that is strange... Which version do you exactly use? `4.1`?

**Comment by gappleto97** [comment]
Im on 4.1 now, but I think it was 4.0 at the time. I know it was 4.* because it was the first time I saw the building height rendered.
